sympy/vector/coordsysrect.py

```diff
@@ -98,6 +213,65 @@ def __new__(cls, name, location=None, rotation_matrix=None,
             location = Vector.zero
             origin = Point(name + '.origin')
 
+        if transformation is None:
+            if parent is None:
```

Comment:
Another thing is `translation`. Without `parent` it doesn't have sense, so we need to distinguish two cases. But I think we can still simplify the code a bit.